tavla/app/components/ContactForm.tsx

```diff
@@ -27,6 +27,13 @@ function ContactForm() {
     )
     const [disabledEmail, setDisabledEmail] = useState(false)
 
+    const setModal: React.Dispatch<SetStateAction<boolean>> = (isVisible) => {
```

Comment:
Er det ikke lettere å ta denne i en `useEffect`? Slipper vi å introdusere en til state som også håndterer åpningen av en modal som kan være litt forvirrende i seg selv siden vi alt har setIsOpen 🤔 

```
    useEffect(() => {
        if (!isOpen) {
            setDisabledEmail(false)
            setFormError(undefined)
        }
    }, [isOpen])
```

Da kan vi også fjerne setDisabledEmail(false) og setFormError(undefined) i submit() funksjonen.

Reply:
Det er sant, var useEffect som var den første tanken som slo meg, men er nok greit å ikke bruke den til å håndtere states. Denne likte jeg! Så veldig clean ut 💅

Reply:
Går det an å sette en key på ContactForm som resetter staten når man er lukker modalen, eller en `key` på selve innholdet som dukker opp når modalen åpnes? 

https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes

Evt en annen måte: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

Reply:
kan undersøke! 🔍

Reply:
prøvde med key, men hadde ingen effekt 😩 fjernet i hvert fall setModal. Det funker med kun resetForm, så det er mindre kode nå enn før. Vi kan undersøke key og andre løsninger etter jul! 🎄